### DISCUSSION TITLE
Inventory management

### DISCUSSION THREAD

**Discussion by brandonkelly** [craftcms/commerce#2286] [Ideas] [answered]
- Manage stock on a per-location basis 
- Allow backorders
- Stock screen tracks adding and deduction of stock.
- Allow Bulk updating of stock.
- Track number of purchasable sales over a period of time.
- Allow reservation of stock while items are in the cart. ?
- Change 'unlimited stock' to 'track stock' ?
- Update stock when manually editing the order on order edit page.
- Update stock option when deleting an order.

**Comment by meinhard-meintjes** [comment]
Hello everyone, I'm also currently trying to figure out how to handle the lack of this feature (manage inventory per site/location). Any updates on the status? If it's still a long time away from being ready, is the easiest/best solution writing a plugin or running a separate instance/DB (in this case 2 instances would be enough) in your opinion? Thanks!

**Comment by jacobrussell** [comment]
@lukeholder @brandonkelly Can you provide any update on the status of this issue?  Is it in your pipeline or still future development?

**Comment by lukeholder** [comment]
@jacobrussell Was and is, but we can't give a timeline on it sorry. Should be in Commerce 4.

**Comment by denisyilmaz** [comment]
Is there a plan to have the backorder-feature arriving in Commerce 3?

**Comment by lukeholder** [comment]
@denisyilmaz no plan in Commerce 3 sorry.

**Comment by denisyilmaz** [comment]
is there any update on a timeline when commerce 4 will be available? some of these features are urgently requested by some projects and it would be very nice to give the project-owners a idea of when these features can be revisited / or if a solution must be found in any other way.

**Comment by jacobrussell** [comment]
@lukeholder Can you provide any update here?  Is this expected at some point this year?  Solid inventory management is a core e-commerce feature that's hard to backfill with custom solutions well.  Would be very helpful to know if we can still expect this with Commerce 4.0, and if that's coming soon.

**Comment by lukeholder** [comment]
@denisyilmaz @jacobrussell we are beginning work on multi-store support, and a stock/inventory system to support each store is a core feature of that. Sorry for the delay, but we are working on it. Sorry I can't give a timeline.

**Comment by CMeldgaard** [comment]
@lukeholder Is creating a custom stock-field the only fix to make backorder possible in Commerce 3 ?

Couldn't this feature be enabled by adding an extra checkbox next to the "Is unlimited", name "Allow backorder", and then simply make af check for $this->allowBackorder in craft\commerce\elementsVariant getLineItemRules ?

Adding this field wouldn't break anything, if its set to false as default (which i think i general should be the default value).

**Comment by mylesthedev** [comment]
👀

**Comment by piotrpog** [comment]
On the dotall 2023 it was announced that next major commerce version will get multi-store support. Does it also mean "Manage stock on a per-location basis"?

**Reply by lukeholder** [reply]
Yes, we will have a lot more to share on it in the near future. In the meantime please let us know what your use-case is for the inventory system and I can let you know if we expect it to meet your needs. Thanks.

**Reply by piotrpog** [reply]
On the commerce project i maintain, each product has table field with stock values. There are two columns - stock number and store symbol. Each row is separate store. This of course is not updated by hand, but using feedme, from external api.

Each store is represented by entry and each store has symbol field, so it can be matched to specific row in stocks table of product and to get stock number from this store for specific product.

When user orders, one of the shipping methods is "getting purchased producs in the store personally". For this method there is additional option (field assigned to order model) - selected store. User can select only the store that has ALL ordered products.

**Comment by wmdhosting** [comment]
I understand..   to limit  Stores to 5  per install.. 

But invetory locations..  limit to 5..   ???

**Reply by lukeholder** [reply]
Hi @wmdhosting 

Please email support@craftcms.com with your project's requirements, and we will do our best to help you!

Luke

**Comment by lukeholder** [comment, marked as answer]
Commerce 5 was just released and it has multi-store and mult-warehouse (inventory location) support!
 
- [x]     Manage stock on a per-location basis
- [x]     Stock screen tracks adding and deduction of stock.
- [x]     Allow Bulk updating of stock.
- [x]     Update stock when manually editing the order on order edit page.
- [x]     Update stock option when deleting an order.
